week-12/homework/index.js

```diff
@@ -0,0 +1,131 @@
+let input = document.querySelector('.input');
+let addBtn = document.querySelector('.add-btn');
+let itemContainer = document.querySelector('.item-container');
+let deleteBtns = document.querySelectorAll('.delete-btn');
+let editBtns = document.querySelectorAll('.edit-btn');
+let currentTaskNum = 0;
+
+//Creates tasklist, takes taskNumber and input value
+let processTaskList = function (taskNum, inputValue) {
+    let newInput = document.createElement("input");
```

Comment:
This whole process below can be optimized to make them a lot less "manual" and easier to maintain in the future.
https://github.com/TaiChiLe/fullstack-learning/pull/2/files#diff-2bbe4e80b92dad33be1498200f3201df02ec5be47bcfe1a0840263cc5c12bd55R10-R48

Check my recommendation below using `innerHTML` and `string`
```
itemContainer.innerHTML =  itemContainer.innerHTML + `
<div class="item d-flex pt-1" item-number="${taskNum}">
    <input class="input-item form-control" type="text" disabled input-item-number="${taskNum}"/>
    <button class="delete-btn btn btn-danger" delte-item-number="${taskNum}">X</button>
    ... you work out the remaining elements yourself
</div>
```

This way, it's a lot faster for you to put together your entire element list and minize the chance of wrong code